lib/rpmsg/rpmsg.c

```diff
@@ -365,7 +365,11 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 		ns_ept.cb = rpmsg_ns_callback;
 		strncpy(ns_ept.name, "NS", sizeof("NS"));
 		status = rpmsg_register_endpoint(rvdev, &ns_ept);
+
 	}
+	/* check if there is message sent from master */
+	if (rpmsg_virtio_get_role(rvdev) == RPMSG_REMOTE)
```

Comment:
Curious why we needed this?

Reply:
if we split slave vdev and master vdev, we don't need this checking, but how to split is still under discussion

Reply:
Sorry, its not the checking I was asking about.  I was asking why we need to call rpmsg_rx_callback.